devel/onetbb/files/patch-onetbb-older-platforms.diff

```diff
@@ -0,0 +1,513 @@
+ # you may not use this file except in compliance with the License.
+@@ -37,7 +37,7 @@ if (CMAKE_OSX_ARCHITECTURES)
+ else()
+     set(_tbb_target_architectures "${CMAKE_SYSTEM_PROCESSOR}")
+ endif()
+-if ("${_tbb_target_architectures}" MATCHES "(x86_64|amd64|AMD64)") # OSX systems are 64-bit only
++if ("${_tbb_target_architectures}" MATCHES "(i.86|x86|x86_64|amd64|AMD64)") # Clang is not supported on PowerPC
+     set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} -mrtm $<$<NOT:$<VERSION_LESS:${CMAKE_CXX_COMPILER_VERSION},12.0>>:-mwaitpkg>)
+ endif()
+ unset(_tbb_target_architectures)
+diff --git a/cmake/compilers/GNU.cmake b/cmake/compilers/GNU.cmake
+index 34c10db0..2496580d 100644
+--- a/cmake/compilers/GNU.cmake
++++ b/cmake/compilers/GNU.cmake
+@@ -36,7 +36,13 @@ if (NOT CMAKE_GENERATOR MATCHES "Ninja" AND NOT CMAKE_CXX_DEPENDS_USE_COMPILER)
+ endif()
+ 
+ # Enable Intel(R) Transactional Synchronization Extensions (-mrtm) and WAITPKG instructions support (-mwaitpkg) on relevant processors
+-if (CMAKE_SYSTEM_PROCESSOR MATCHES "(AMD64|amd64|i.86|x86)")
++# Like with Apple Clang case, use CMAKE_OSX_ARCHITECTURES if it is defined. Defaulting to CMAKE_SYSTEM_PROCESSOR breaks build on Rosetta.
++if (CMAKE_OSX_ARCHITECTURES)
++    set(_tbb_target_architectures "${CMAKE_OSX_ARCHITECTURES}")
++else()
++    set(_tbb_target_architectures "${CMAKE_SYSTEM_PROCESSOR}")
++endif()
++if ("${_tbb_target_architectures}" MATCHES "(AMD64|amd64|i.86|x86)")
+     set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} -mrtm $<$<AND:$<NOT:$<CXX_COMPILER_ID:Intel>>,$<NOT:$<VERSION_LESS:${CMAKE_CXX_COMPILER_VERSION},11.0>>>:-mwaitpkg>)
+ endif()
+ 
+diff --git a/include/oneapi/tbb/concurrent_queue.h b/include/oneapi/tbb/concurrent_queue.h
+index 24659715..6ec705c2 100644
+--- a/include/oneapi/tbb/concurrent_queue.h
++++ b/include/oneapi/tbb/concurrent_queue.h
+@@ -1,5 +1,5 @@
+ /*
+-    Copyright (c) 2005-2022 Intel Corporation
++    Copyright (c) 2005-2023 Intel Corporation
+ 
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+@@ -241,8 +241,8 @@ namespace r1 {
+     TBB_EXPORT std::uint8_t* __TBB_EXPORTED_FUNC allocate_bounded_queue_rep( std::size_t queue_rep_size );
+     TBB_EXPORT void __TBB_EXPORTED_FUNC deallocate_bounded_queue_rep( std::uint8_t* mem, std::size_t queue_rep_size );
+     TBB_EXPORT void __TBB_EXPORTED_FUNC abort_bounded_queue_monitors( concurrent_monitor* monitors );
+-    TBB_EXPORT void __TBB_EXPORTED_FUNC notify_bounded_queue_monitor( concurrent_monitor* monitors, std::size_t monitor_tag
+-                                                            , std::size_t ticket );
++    TBB_EXPORT void __TBB_EXPORTED_FUNC notify_bounded_queue_monitor( concurrent_monitor* monitors, std::size_t monitor_tag,
++                                                            std::size_t ticket );
```

Comment:
This seems unnecessary. Best to keep the patch minimal to what's required.

Reply:
Sure, will remove that now.